#  Film Industry Analysis for Business Expansion 

## 🎬 **Introduction**  
As part of a company's expansion into the entertainment industry, they are exploring the creation of a new movie studio. However, with no prior experience in film production, it is critical to make data-driven decisions about the types of movies to develop.  

In this project, I will analyze **movie industry data** to answer key business questions such as:  
- **Which movie genres tend to receive the highest ratings?**  
- **Which genres have the highest return on investment (ROI)?**  
- **Are certain directors more likely to deliver high foreign box office success?**  
- **Which film languages are associated with higher popularity and profitability?**  

The goal is to uncover patterns and insights that will help guide the company's decisions on what types of films to produce for maximum commercial success.


## 【1】**Data Exploration & Cleaning**  

### **rt.ratings and genre data prep**

#### **Rotten Tomatoes Movies Dataset Description**

The **Rotten Tomatoes Movies** dataset provides detailed information about a wide range of movies, collected from the Rotten Tomatoes website.  
It contains metadata that can be used to analyze movie characteristics, trends, audience perceptions, and more.

---

#### **Key features**
- **movie_title** The title of the movie as listed on Rotten Tomatoes.
- **genres** A comma-separated list of genres assigned to the movie (e.g., Drama, Comedy, Action).
- **audience_rating** The average rating given by the audience, typically scored from 0 to 100.
- **tomatometer_rating** The percentage of positive critics rating


---

#### **Key Points**
- Some movies belong to **multiple genres** (e.g., "Action, Comedy, Drama").
- Not all entries have complete information; there might be **missing values** for some fields.
- The audience rating reflects the **general public's** reception rather than professional critics' reviews.
- Some text fields (like movie titles and genres) may have **inconsistent formatting** (e.g., upper vs lower case).

---




### **step 1: reading the dataset**

In [5]:
#importing the necessary libraries 
import numpy as np
import pandas as pd

We import essential Python libraries:
- **NumPy** for numerical operations.
- **Pandas** for data manipulation and analysis.


In [6]:
# loading the dataset
rt_movies=pd.read_csv(r"C:\Users\HomePC\Desktop\phase_2-project\dsc-phase-1-project-template\rotten_tomatoes_movies.csv")
rt_movies.sample(5)

,rotten_tomatoes_link,movie_title,movie_info,critics_consensus,content_rating,genres,directors,authors,actors,original_release_date,...,production_company,tomatometer_status,tomatometer_rating,tomatometer_count,audience_status,audience_rating,audience_count,tomatometer_top_critics_count,tomatometer_fresh_critics_count,tomatometer_rotten_critics_count
9878,m/lurking_fear,Lurking Fear,Five gutsy townspeople unite to eradicate a gh...,NaN,R,"Cult Movies, Horror",C. Courtney Joyner,"C. Courtney Joyner, Adolfo Bartolli","Jon Finch, Blake Bailey, Ashley Laurence, Ashl...",1994-07-27,...,Full Moon,Rotten,20.0,5.0,Spilled,13.0,249.0,0,1,4
16448,m/transit_2012,Transit,A family's vacation goes horribly wrong when b...,NaN,R,"Drama, Mystery & Suspense, Special Interest",Antonio Negret,Michael Gilvary,"Jim Caviezel, James Frain, Harold Perrineau, D...",2012-05-11,...,After Dark Films,Rotten,50.0,8.0,Spilled,30.0,756.0,3,4,4
1107,m/1072457-chain_reaction,Chain Reaction,"At a Chicago university, a research team that ...",Ironic given the scientific breakthrough at th...,PG-13,"Action & Adventure, Mystery & Suspense",Andrew Davis,"Josh Friedman, Rick Seaman, Michael Bortman, J...","Keanu Reeves, Morgan Freeman, Rachel Weisz, Fr...",1996-08-02,...,Twentieth Century Fox Home Entertainment,Rotten,18.0,33.0,Spilled,27.0,52509.0,10,6,27
10628,m/mothers_day,Mother's Day,Three women meet demented brothers who live in...,NaN,R,Horror,Charles Kaufman,"Warren Leight, Charles Kaufman","Nancy Hendrickson, Deborah Luce, Tiana Pierce,...",1980-09-01,...,United Film Distribution Compa,Rotten,42.0,12.0,Spilled,38.0,2735.0,3,5,7
14192,m/t2_trainspotting,T2 Trainspotting,"First there was an opportunity, then there was...","T2 Trainspotting adds an intoxicating, emotion...",R,Drama,Danny Boyle,John Hodge,"Ewan McGregor, Ewen Bremner, Jonny Lee Miller,...",2017-03-17,...,TriStar Pictures,Certified-Fresh,81.0,249.0,Upright,78.0,16203.0,48,202,47


We load the movie data from a CSV file into a pandas DataFrame called `rt_movies`.  
A sample of 5 random rows is displayed to understand the structure of the dataset.


#### **step 2 extracting the relevant columns and exploding the genre column**


We select only three columns: `movie_title`, `genres`,`tomatometer_rating` and `audience_rating` because these are the columns we will need in our analysis.
- The `genres` column contains multiple genres separated by commas.  
- We **split** this column into lists so that each genre can later be separated into individual rows.


In [10]:
#extracting the relevant columns
df_rt=rt_movies[["movie_title","genres","audience_rating","tomatometer_rating"]]
# Convert the comma-separated string into a list
df_rt['genres'] = df_rt['genres'].str.split(', ')

# exploding the genres
rt_movies_exploded = df_rt.explode('genres').reset_index(drop=True)
rt_movies_exploded.head()



C:\Users\HomePC\AppData\Local\Temp\ipykernel_11136\2531300135.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_rt['genres'] = df_rt['genres'].str.split(', ')


,movie_title,genres,audience_rating,tomatometer_rating
0,Percy Jackson & the Olympians: The Lightning T...,Action & Adventure,53.0,49.0
1,Percy Jackson & the Olympians: The Lightning T...,Comedy,53.0,49.0
2,Percy Jackson & the Olympians: The Lightning T...,Drama,53.0,49.0
3,Percy Jackson & the Olympians: The Lightning T...,Science Fiction & Fantasy,53.0,49.0
4,Please Give,Comedy,64.0,87.0


We **explode** the `genres` column:
- Each genre becomes a separate row.
- This transforms one-to-many relationships into one-to-one, making it easier to analyze by individual genre.
- After exploding, we reset the index to clean up the DataFrame.


In [8]:
#comparing the number of rows for main df and the exploded df 
print(f"main:{rt_movies.shape[0]} rows")
print(f"exploded:{rt_movies_exploded.shape[0]} rows")


main:17712 rows
exploded:39388 rows


We compare the number of rows in:
- The original DataFrame
- The exploded DataFrame

An increase in rows confirms that multiple genres per movie were successfully separated into multiple rows.


In [9]:
#seeing if the number of unique movie_title values matches with the main
print(f"exploded: {len(rt_movies_exploded['movie_title'].unique())} unique values")
print(f"main df: {len(rt_movies['movie_title'].unique())} unique values")


exploded: 17106 unique values
main df: 17106 unique values


We verify that the number of unique `movie_title` entries remains consistent between the original and exploded datasets.
This ensures that exploding the genres did not accidentally duplicate or lose any movies.



### **step 3: summarizing the dataset**

Summarizing a DataFrame is essential to **quickly understand the dataset** and spot issues early.  
It helps in:

- **Understanding structure**: Number of rows, columns, and their types.
- **Detecting missing values**: Finding gaps that could affect analysis.
- **Checking data types**: Ensuring columns are correctly formatted for operations.
- **Assessing memory usage**: Optimizing performance for large datasets.
- **Identifying potential problems**: Catching inconsistencies like wrong formats or placeholders.
- **Guiding cleaning steps**: Deciding what transformations are needed.
- **Saving time**: Providing a quick overview instead of manual checking.

---


In [11]:
# summary of the explode dataset
rt_movies_exploded.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 39388 entries, 0 to 39387
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   movie_title         39388 non-null  object 
 1   genres              39369 non-null  object 
 2   audience_rating     38899 non-null  float64
 3   tomatometer_rating  39296 non-null  float64
dtypes: float64(2), object(2)
memory usage: 1.2+ MB


the dataframe has **39388 rows and 3 columns**. `genres`,`audience_rating`, and `tomatometer_rating` **have missing values** but the missing values  are not significant. `audience rating` and `tomatometer rating` columns are **numerical** while `movie title` and the `genres` colums are **categorical**.
all the column values have the same **data types** as the ones we expect them to have

### **step 4: description of the dataframe and checking for outliers**

#### **dataframe description**

Describing a DataFrame gives a **statistical summary** of numerical columns, providing **deeper insights** into the data distribution and behaviour.

It helps in:
- Understanding Data Distribution
- Detecting Outliers
- Checking Data Variability
- Validating Data Accuracy
- Supporting Feature Engineering
- Assessing Skewness




In [12]:
##describing the numerical column
rt_movies_exploded.describe()


,audience_rating,tomatometer_rating
count,38899.000000,39296.000000
mean,61.542945,62.107619
std,20.373430,28.047419
min,0.000000,0.000000
25%,46.000000,40.000000
50%,64.000000,67.000000
75%,78.000000,86.000000
max,100.000000,100.000000


This table compares the distribution of **Audience Ratings** and **Tomatometer Ratings** from a sample of movie reviews. The statistics highlight differences in central tendency, spread, and skew between general audience scores and critic-aggregated ratings.

- **Sample Size**:  
  - Audience Ratings: **38,899** reviews  
  - Tomatometer Ratings: **39,296** reviews  
- **Average Ratings**:  
  - Audiences tend to rate slightly lower (**61.54**) compared to critics (**62.11**).  
- **Variability**:  
  - Critics' ratings exhibit higher variability (**std = 28.05**) than audience ratings (**std = 20.37**).  
- **Distributions**:  
  - Both metrics range from **0–100**, but critics' ratings are more polarized:  
    - 25% of critic scores are ≤ **40**, while 75% of audience scores are ≤ **78**.  
    - Median critic ratings (**67**) are higher than audience medians (**64**).  
  

#### **checking for outliers**

Using the Interquartile Range (IQR) method, we check for outliers:
- **IQR = Q3 - Q1**
- Any data point below **Q1 - 1.5 * IQR** or above **Q3 + 1.5 * IQR** is considered an outlier.

We apply this logic to:
- `audience_rating`
- `tomatometer_rating`

Detecting and handling outliers is crucial for reliable statistical analysis.


In [14]:
#checking for outliers
def check_outliers(q1,q3,min,max):
    iqr=q3-q1
    upper_bound=q3+(1.5*iqr)
    lower_bound=q1-(1.5*iqr)
    if (min>lower_bound) & (max<upper_bound):
     return("There are no outliers")
    else:
     return("There are outliers")
# checking in the audience_rating column
print(f"{check_outliers(q1=46,q3=78,min=0,max=100)} in the audience rating column")
# checking in the tomatometer_rating column
print(f"{check_outliers(q1=40,q3=86,min=0,max=100)} in the tomatometer rating column")


There are no outliers in the audience rating column
There are no outliers in the tomatometer rating column


### **step 5 : checking and handling missing values**

Understanding missing data is important to Decide whether to remove or impute missing values which will maintain data integrity for further analysis.


#### **checking for missing values**

In [18]:
# checking for missing values in terms of percentage
(rt_movies_exploded.isna().sum()/len(rt_movies_exploded)*100)

movie_title           0.000000
genres                0.048238
audience_rating       1.241495
tomatometer_rating    0.233574
dtype: float64

`genres`,`audience_rating` and `tomatometer_rating` have missing values in the respective columns, but the missing values are insignificant compared to the total number of entries.
We drop all the rows with missing values since dropping them won't impact our analyses.

#### **handling missing values**

In [19]:
# dropping the rows which have empty values under the columns
rt_movies_exploded_cleaned=rt_movies_exploded.dropna(axis=0,subset="genres")
rt_movies_exploded_cleaned.dropna(axis=0,subset="audience_rating",inplace=True)
rt_movies_exploded_cleaned.dropna(axis=0,subset="tomatometer_rating",inplace=True)
# checking if there is still any missing values left
rt_movies_exploded_cleaned.isna().sum()

C:\Users\HomePC\AppData\Local\Temp\ipykernel_11136\1016391485.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  rt_movies_exploded_cleaned.dropna(axis=0,subset="audience_rating",inplace=True)
C:\Users\HomePC\AppData\Local\Temp\ipykernel_11136\1016391485.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  rt_movies_exploded_cleaned.dropna(axis=0,subset="tomatometer_rating",inplace=True)


movie_title           0
genres                0
audience_rating       0
tomatometer_rating    0
dtype: int64

After dropping the missing values, the dataframe now does not have any missing values

In [20]:
## checking for the number of rows that reduced after handling missing values
reduced=rt_movies_exploded_cleaned.shape[0]-rt_movies_exploded.shape[0]
percent_reduced=round((reduced/len(rt_movies_exploded))*100)
print(f"percentage reduced={percent_reduced} percent")



percentage reduced=-1 percent


after cleaning the dataframe the percentage of the rows that reduced from the original dataframe is 1 percent hence it will not really affect our analyses 

#### **step 6: checking for placeholders**

checking and handling placeholders improves the integrity of the data

In [21]:
# checking for placeholders in the genres column
rt_movies_exploded_cleaned["genres"].value_counts()

genres
Drama                        9432
Comedy                       5593
Action & Adventure           3526
Mystery & Suspense           3502
Art House & International    2476
Horror                       2023
Documentary                  1960
Romance                      1897
Science Fiction & Fantasy    1845
Classics                     1647
Special Interest             1211
Kids & Family                1109
Musical & Performing Arts     989
Animation                     624
Western                       301
Television                    265
Sports & Fitness              175
Cult Movies                   104
Faith & Spirituality           79
Gay & Lesbian                  76
Anime & Manga                  19
Name: count, dtype: int64

there was no placeholder in the genres column.
since the minimum value in the audience_rating and tomatometer_rating is 0 in each column and the maximum value is 100, we can conclude that they have no placeholders.

### genres and ROI data prep
#### Vincent can make edits to this markdown cell appropriately in your own branch, 
as you guys can see we have mixed both data exploration and data cleaning into one so as you are prepping the data for analysis use appropriate markdown cells where necessary to explain yourself and also do not edit the other cells without your name in it as it will raise merge conflicts when merging all our code together
so you can add as many markdown and code cells in this space before the next guys name.
also if you want to edit anything in the document make note of it and dont change it in your branch just work on the assigned code task in your branch overall fine tuning we will come to that later.
this markdown cell is just as a reminder when you start the work you can remove everything here and make appropriate subtitles and markdown cells for your code

### language and popularity data prep
#### Chris can make edits to this markdown cell appropriately in your own branch, 
as you guys can see we have mixed both data exploration and data cleaning into one so as you are prepping the data for analysis use appropriate markdown cells where necessary to explain yourself and also do not edit the other cells without your name in it as it will raise merge conflicts when merging all our code together
so you can add as many markdown and code cells in this space before the next guys name.
also if you want to edit anything in the document make note of it and dont change it in your branch just work on the assigned code task in your branch overall fine tuning we will come to that later.
this markdown cell is just as a reminder when you start the work you can remove everything here and make appropriate subtitles and markdown cells for your code

### directors and foreign gross data prep
#### VIhaan can make edits to this markdown cell appropriately in your own branch, 

## 【2】**Data Analysis**  

#### *1. Data Analysis*  

#### *2. Hypothesis Testing*  

## 【3】**Business Recommendations**  

## 【4】**Limitations of Our Analysis**  